clippy_lints/src/utils/mod.rs

```diff
@@ -50,17 +50,68 @@ use rustc_lint::{LateContext, Level, Lint, LintContext};
 use rustc_middle::hir::map::Map;
 use rustc_middle::ty::subst::{GenericArg, GenericArgKind};
 use rustc_middle::ty::{self, layout::IntegerExt, Ty, TyCtxt, TypeFoldable};
+use rustc_session::Session;
 use rustc_span::hygiene::{ExpnKind, MacroKind};
 use rustc_span::source_map::original_sp;
 use rustc_span::sym as rustc_sym;
 use rustc_span::symbol::{self, kw, Symbol};
 use rustc_span::{BytePos, Pos, Span, DUMMY_SP};
 use rustc_target::abi::Integer;
 use rustc_trait_selection::traits::query::normalize::AtExt;
+use semver::{Version, VersionReq};
 use smallvec::SmallVec;
 
 use crate::consts::{constant, Constant};
 
+pub fn parse_msrv(msrv: &str, sess: Option<&Session>, span: Option<Span>) -> Option<VersionReq> {
+    if let Ok(version) = VersionReq::parse(msrv) {
+        return Some(version);
+    } else {
+        if let Some(sess) = sess {
+            if let Some(span) = span {
+                sess.span_err(span, &format!("`{}` is not a valid Rust version", msrv));
+            }
+        }
+    }
```

Comment:
```suggestion
    } else if let Some(sess) = sess {
        if let Some(span) = span {
            sess.span_err(span, &format!("`{}` is not a valid Rust version", msrv));
        }
    }
```

CI says this `else { if let` is redundant.

```text

error: this `else { if .. }` block can be collapsed
  --> clippy_lints/src/utils/mod.rs:70:12
   |
70 |       } else {
   |  ____________^
71 | |         if let Some(sess) = sess {
72 | |             if let Some(span) = span {
73 | |                 sess.span_err(span, &format!("`{}` is not a valid Rust version", msrv));
74 | |             }
75 | |         }
76 | |     }
   | |_____^
   |
   = note: `-D clippy::collapsible-if` implied by `-D clippy::all`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#collapsible_if
help: collapse nested if block
   |
70 |     } else if let Some(sess) = sess {
71 |         if let Some(span) = span {
72 |             sess.span_err(span, &format!("`{}` is not a valid Rust version", msrv));
73 |         }
74 |     }
   |
```

Reply:
@taiki-e yep, I'm taking a look at the dogfood test failure logs